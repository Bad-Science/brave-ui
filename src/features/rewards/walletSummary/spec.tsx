/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import WalletSummary, { Props } from './index'

const props = {
  report: {
    grant: {tokens: 10, converted: 0.25},
    ads: {tokens: 10, converted: 0.25},
    contribute: {tokens: 10, converted: 0.25},
    donation: {tokens: 2, converted: 0.25},
    tips: {tokens: 19, converted: 5.25},
    total: {tokens: 19, converted: 5.25},
  },
  onActivity: ()=>{}
}

describe('WalletSummary tests', () => {
  const baseComponent = (props: Props) => <WalletSummary id='empty' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent(props)
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent(props))
      const assertion = wrapper.find('#empty').length
      expect(assertion).toBe(1)
    })
  })
})
