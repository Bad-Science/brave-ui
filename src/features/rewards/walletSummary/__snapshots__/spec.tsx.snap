// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletSummary tests basic tests matches the snapshot 1`] = `
.c0 {
  padding: 0px;
  background: inherit;
}

.c1 {
  padding: 14px;
  font-family: Poppins,sans-serif;
}

.c2 {
  font-size: 14px;
  font-weight: 600;
  line-height: 1.57;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  color: #a1a8f2;
  text-transform: uppercase;
}

.c3 {
  font-size: 28px;
  font-weight: 300;
  line-height: 0.79;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  color: #4c54d2;
  margin: 4px 0 26px;
  text-transform: uppercase;
}

.c14 {
  font-size: 12px;
  color: #686978;
  margin-top: 26px;
  text-align: center;
  padding: 0;
  border: none;
  background: none;
  width: 100%;
  cursor: pointer;
}

.c15 {
  vertical-align: middle;
  margin-right: 11px;
  width: 22px;
  height: 24px;
  color: #A1A8F2;
  display: inline-block;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #d0d6dc;
  border-top: none;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  font-family: Poppins,sans-serif;
}

.c5 {
  font-size: 14px;
  line-height: 2.79;
  color: #4b4c5c;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 60%;
  -ms-flex-preferred-size: 60%;
  flex-basis: 60%;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  text-align: right;
}

.c7 {
  --tokens-value-color: #c12d7c;
  --tokens-text-size: 12px;
  --tokens-token-size: 14px;
  --tokens-tokenNum-size: 16px;
}

.c12 {
  --tokens-value-color: #9752cb;
  --tokens-text-size: 12px;
  --tokens-token-size: 14px;
  --tokens-tokenNum-size: 16px;
}

.c13 {
  --tokens-value-color: #696fdc;
  --tokens-text-size: 12px;
  --tokens-token-size: 14px;
  --tokens-tokenNum-size: 16px;
}

.c8 {
  font-family: Poppins,sans-serif;
  font-weight: 300;
  line-height: 1.4;
  color: #686978;
  display: inline-block;
}

.c9 {
  color: var(--tokens-value-color);
  font-size: var(--tokens-tokenNum-size);
  font-weight: 500;
}

.c11 {
  color: #9E9FAB;
  font-size: var(--tokens-text-size);
  font-family: Muli,sans-serif;
  line-height: 1.29;
  display: inline-block;
  margin-left: 10px;
}

.c10 {
  font-size: var(--tokens-token-size);
  font-weight: 300;
  display: inline-block;
  margin-left: 4px;
}

.c16 {
  width: 100%;
  height: 100%;
  fill: currentColor;
}

<div
  className="c0"
  id="empty"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      MISSING: rewardsSummary
    </div>
    <div
      className="c3"
    >
      MISSING: monthOct
       
      2018
    </div>
    <div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          MISSING: tokenGrant
        </div>
        <div
          className="c6"
        >
          <span
            className="c7"
            color="earnings"
            size="normal"
          >
            <span
              className="c8"
            >
              <span
                className="c9"
              >
                
                10.0
              </span>
              <span
                className="c10"
              >
                BAT
              </span>
            </span>
            <span
              className="c11"
            >
              0.25
               
              USD
            </span>
          </span>
        </div>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          MISSING: earningsAds
        </div>
        <div
          className="c6"
        >
          <span
            className="c7"
            color="earnings"
            size="normal"
          >
            <span
              className="c8"
            >
              <span
                className="c9"
              >
                
                10.0
              </span>
              <span
                className="c10"
              >
                BAT
              </span>
            </span>
            <span
              className="c11"
            >
              0.25
               
              USD
            </span>
          </span>
        </div>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          MISSING: rewardsContribute
        </div>
        <div
          className="c6"
        >
          <span
            className="c12"
            color="contribute"
            size="normal"
          >
            <span
              className="c8"
            >
              <span
                className="c9"
              >
                -
                10.0
              </span>
              <span
                className="c10"
              >
                BAT
              </span>
            </span>
            <span
              className="c11"
            >
              0.25
               
              USD
            </span>
          </span>
        </div>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          MISSING: recurringDonations
        </div>
        <div
          className="c6"
        >
          <span
            className="c13"
            color="donation"
            size="normal"
          >
            <span
              className="c8"
            >
              <span
                className="c9"
              >
                -
                2.0
              </span>
              <span
                className="c10"
              >
                BAT
              </span>
            </span>
            <span
              className="c11"
            >
              0.25
               
              USD
            </span>
          </span>
        </div>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          MISSING: oneTimeDonation
        </div>
        <div
          className="c6"
        >
          <span
            className="c13"
            color="donation"
            size="normal"
          >
            <span
              className="c8"
            >
              <span
                className="c9"
              >
                -
                19.0
              </span>
              <span
                className="c10"
              >
                BAT
              </span>
            </span>
            <span
              className="c11"
            >
              5.25
               
              USD
            </span>
          </span>
        </div>
      </div>
    </div>
    <button
      className="c14"
      onClick={[Function]}
    >
      <span
        className="c15"
      >
        <svg
          aria-hidden="true"
          className="c16"
          focusable="false"
          viewBox="0 0 32 32"
        >
          <path
            d="M25.571 6C26.91 6 28 7.101 28 8.455V17c0 .552-.447 1-1 1-.553 0-1-.448-1-1V8.455c0-.247-.196-.455-.429-.455H8.429C8.196 8 8 8.208 8 8.455V12h5.001c.553 0 1 .448 1 1v4c0 .552-.447 1-1 1H8v3.545c0 .247.196.455.429.455H17c.553 0 1 .448 1 1s-.447 1-1 1H8.429C7.09 24 6 22.899 6 21.545V18H5c-.553 0-1-.448-1-1v-4c0-.552.447-1 1-1h1V8.455C6 7.101 7.09 6 8.429 6h17.142zM6 16h6.001v-2H6v2zm16 3c-.553 0-1-.448-1-1s.447-1 1-1h1c.553 0 1 .448 1 1s-.447 1-1 1h-1zm2 5c.553 0 1 .448 1 1s-.447 1-1 1h-2c-.553 0-1-.448-1-1s.447-1 1-1v-2c-.553 0-1-.448-1-1s.447-1 1-1h1c.553 0 1 .448 1 1v3z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      MISSING: viewMonthly
    </button>
  </div>
</div>
`;
